dev/com.ibm.ws.jaxws.2.2.webcontainer_fat_extended/test-applications/testHandlerProvider/src/com/ibm/samples/jaxws/testhandlerprovider/handler/TestSOAPHandlerAddedElement.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.samples.jaxws.testhandlerprovider.handler;
+
+import java.util.Set;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.xml.namespace.QName;
+import javax.xml.soap.Node;
+import javax.xml.soap.SOAPException;
+import javax.xml.ws.handler.MessageContext;
+import javax.xml.ws.handler.soap.SOAPHandler;
+import javax.xml.ws.handler.soap.SOAPMessageContext;
+
```

Comment:
Can you add code comments to this class file. Something like 

```suggestion
// This handler adds an optional SOAPElement on outgoing responses from SayHelloServiceWithHandlerImpl"
```